financialaid/views.py

```diff
@@ -1,24 +1,229 @@
+            "reactGaDebug": settings.REACT_GA_DEBUG,
+            "authenticated": not self.request.user.is_anonymous(),
+            "edx_base_url": settings.EDXORG_BASE_URL,
+        }
+        context["js_settings_json"] = json.dumps(js_settings)
+        return context
+
+    def get_queryset(self):
+        """
+        Gets queryset for ListView to return to view
+        """
+        # Filter by availability
+        financial_aids = FinancialAid.objects.filter(
+            tier_program__program__live=True,
+            tier_program__program__financial_aid_availability=True
+        )
+
+        # Filter by program (self.program set in test_func())
+        financial_aids = financial_aids.filter(
+            tier_program__program=self.program,
+        )
+
+        # Filter by status
+        self.selected_status = self.kwargs.get("status", None)
+        if self.selected_status is None or self.selected_status not in FinancialAidStatus.ALL_STATUSES:
+            self.selected_status = FinancialAidStatus.PENDING_MANUAL_APPROVAL
+        financial_aids = financial_aids.filter(status=self.selected_status)
+
+        # Sort by field
+        self.sort_field = self.request.GET.get("sort_by", self.default_sort_field)
+        if self.sort_field.startswith("-"):
+            self.sort_field = self.sort_field[1:]
+            # Defined above: self.sort_direction = ""
+            self.sort_direction = "-"
+        if self.sort_field not in self.sort_fields:
+            self.sort_field = self.default_sort_field
+            self.sort_direction = ""
+        financial_aids = financial_aids.order_by(
+            "{sort_direction}{sort_field}".format(
+                sort_direction=self.sort_direction,
+                sort_field=self.sort_field_mappings.get(self.sort_field, self.sort_field)
+            )
+        )
+
+        # Add adjusted cost and TierPrograms valid for selection for each FinancialAid object
+        # Note: iterating through this queryset, if large, can be very costly in terms of memory/time,
+        # but it is done here to simplify the templates.
+        # Get program ids and associated tier programs
+        program_ids = Program.objects.filter(
+            live=True,
+            financial_aid_availability=True
+        ).values_list(
+            "id",
+            flat=True
+        )
+        # Get course price to calculate adjusted cost
+        # Note: This implementation of retrieving a course price is a naive lookup that assumes
+        # all course runs and courses will be the same price for the foreseeable future,
+        # irrespective of the program. Therefore we can just take the price from any currently
+        # enroll-able course run.
+        course_price = CoursePrice.objects.filter(
+            is_valid=True,
+            course_run__in=CourseRun.objects.filter(
+                course__program_id=self.program.id
+            ).filter(
+                CourseRun.get_active_enrollment_queryset()
```

Comment:
Thanks. I think since we are operating under the assumption that prices are the same for each course run in a program, they should be the same no matter whether the run is under active enrollment or not. At some point in the future we'll look into simplifying this.